infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/engine/statement/SQLStatementBinderContext.java

```diff
@@ -47,6 +49,8 @@ public final class SQLStatementBinderContext {
     
     private final SQLStatement sqlStatement;
     
+    private final Set<CaseInsensitiveString> commonTableExpressionsSegmentsUniqueAliases = new HashSet<>();
```

Comment:
Please use `Collection<CaseInsensitiveString>` to replace `Set<CaseInsensitiveString>`.

Reply:
Besides, dou you think CaseInsensitiveSet is better?

Reply:
Now I am using `Collection<String> commonTableExpressionsSegmentsUniqueAliases = new CaseInsensitiveSet<>();`

CaseInsensitiveSet<String> will be better as it ease the overhead of creating the object in case of Set<CaseInsensitiveString>